pkg/frontend/querymiddleware/astmapper/sharding.go

```diff
@@ -44,44 +44,42 @@ func (lbl *queryShardLabeller) GetParams(_ int) map[string]string {
 }
 
 // NewQueryShardSummer instantiates an ASTMapper which will fan out sum queries by shard.
-func NewQueryShardSummer(shards int, inspectOnly bool, squasher Squasher, logger log.Logger, stats *MapperStats) ASTMapper {
-	return NewShardSummerWithLabeller(shards, inspectOnly, squasher, logger, stats, newQueryShardLabeller(shards))
+func NewQueryShardSummer(shards int, squasher Squasher, logger log.Logger, stats *MapperStats) ASTMapper {
+	return NewShardSummerWithLabeller(shards, squasher, logger, stats, newQueryShardLabeller(shards))
 }
 
-func NewShardSummerWithLabeller(shards int, inspectOnly bool, squasher Squasher, logger log.Logger, stats *MapperStats, labeller ShardLabeller) ASTMapper {
-	summer := newShardSummer(shards, inspectOnly, squasher, logger, stats, labeller)
+func NewShardSummerWithLabeller(shards int, squasher Squasher, logger log.Logger, stats *MapperStats, labeller ShardLabeller) ASTMapper {
+	summer := newShardSummer(shards, squasher, logger, stats, labeller)
 	return NewASTExprMapper(summer)
 }
 
 type shardSummer struct {
```

Comment:
Nit, and not directly related to the PR:

The comment for NewQueryShardSummer says:
// NewQueryShardSummer instantiates an ASTMapper which will fan out sum queries by `shard.`
That made me think it should only handle sum(...) queries.
but it also doing stuff with avg and other aggregations. 
Am I missing something?